src/types.rs

```diff
@@ -233,17 +241,36 @@ pub mod consent {
 // pub type AeadNonce = [u8; 12];
 // pub type AeadTag = [u8; 16];
 
-// The "ClientContext" struct is the closest equivalent to a PCB that Trussed
+/// The context for a syscall (per client).
+///
+/// The context stores the state used by the standard syscall implementations, see
+/// [`CoreContext`][].  Additionally, backends can define a custom context for their syscall
+/// implementations.
+pub struct Context<B> {
+    pub core: CoreContext,
+    pub backends: B,
+}
+
+impl<B: Default> From<CoreContext> for Context<B> {
+    fn from(core: CoreContext) -> Self {
+        Self {
+            core,
+            backends: B::default(),
+        }
+    }
+}
+
+// The "CoreContext" struct is the closest equivalent to a PCB that Trussed
 // currently has. Trussed currently uses it to choose the client-specific
 // subtree in the filesystem (see docs in src/store.rs) and to maintain
 // the walker state of the directory traversal syscalls.
-pub struct ClientContext {
+pub struct CoreContext {
```

Comment:
Shouldn't this be non-exhaustive in light of other discussions?

Reply:
I’ll add it to #71.  It has to be part of the public API so that the backends have access to it.